lighthouse-core/gather/driver.js

```diff
@@ -918,6 +911,9 @@ class Driver {
           maxWaitMs);
     }
 
+    // Awaiting so that errors can be properly processed.
+    await waitforPageNavigateCmd;
```

Comment:
there's nothing in _endNetworkStatusMonitoring we need, right?

Reply:
> there's nothing in _endNetworkStatusMonitoring we need, right?

it returns the `finalUrl` after redirects, which is used for updating `passContext` (and eventually the artifact).

A little too convoluted but somewhat difficult to fix since `networkRecords` don't entirely exist by that point.